[Cheak Han Wei] iP

hanweic53 opened the pull request.

Commit:
Add Gradle support

hanweic53 committed:
Completed Level-1: Greet, Echo, Exit

hanweic53 committed:
Completed Level-2: Add, List

hanweic53 committed:
Completed Level-3: Mark as Done

hanweic53 committed:
Completed Level-4: ToDos, Events, Deadlines

hanweic53 committed:
Completed A-TextUiTesting

hanweic53 committed:
Completed Level-5: Handle Errors

Updated A-TextUiTesting to test for errors

hanweic53 committed:
Completed Level-6: Delete

Updated A-TextUiTesting to test for Delete

hanweic53 committed:
Add Level-7: Save

Tasks are not saved and only stored in temporary variables.
Users start the program with clean data.
We will save the data in the hard disk and load the data when Duke starts up.

hanweic53 committed:
Add Level-8: Dates and Times

Dates were stored as Strings.
Not very meaningful to do so.
Parsing dates and storing them as LocalTimeDate Objects allow us to perform
more meaningful operations that deal with time and also to standardise
users' input.

hanweic53 committed:
Merge branch 'branch-Level-7' into master

hanweic53 committed:
Merge branch 'branch-Level-8' into master

hanweic53 committed:
Refactor Level-8: Dates and Times

Code was merged from branch-Level-8 to master branch.
Errors were thrown when Duke was ran.
Refactored Level-8.

hanweic53 committed:
Add A-MoreOOP: Use More OOP

Made the program more OOP by extracting Ui, Parser, TaskList and the
Commands classes.

hanweic53 committed:
Add A-Packages: Organize into Packages

Working directory is very cluttered.
Put the files into different packages.

hanweic53 committed:
Add A-JUnit: Add JUnit Tests

Add JUnit for testing code.
Added methods to test methods in AddCommand, DeleteCommand and
ExitCommand classes.

hanweic53 committed:
Add A-JavaDoc: JavaDoc

Added JavaDocs to Commands classes and Duke.
JavaDocs helps external users to understand our code better.

hanweic53 committed:
Add A-CodingStandard: Follow the Coding Standard

Tweak the code to comply with the SE-EDU coding standard.

hanweic53 committed:
Add Level-9: Find

Give users a way to find a task by searching for a keyword.

hanweic53 committed:
Merge branch 'branch-A-JavaDoc' into master

hanweic53 committed:
Add Level-9: Find

Missed out the FindCommand class in the previous commit.

hanweic53 committed:
Merge branch 'branch-CodingStandard' into master

hanweic53 committed:
Merge branch 'branch-Level-9' into master

hanweic53 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into master

hanweic53 committed:
Add A-Gradle: Use Gradle

Automate project builds using Gradle.
Gradle can run Unit tests, as well as run the Duke program.

hanweic53 committed:
Merge branch 'branch-A-Gradle' into master

hanweic53 committed:
Add A-CheckStyle: Use CheckStyle

To detect coding style violations.

hanweic53 committed:
Merge branch 'branch-A-CheckStyle' into master

hanweic53 committed:
Add Level-10: GUI

Simple GUI implemented for Duke.

hanweic53 committed:
Add Level-10: Refactor GUI using fxml

Implemented GUI using fxml.

Zhang-Yunjie commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,164 @@
+
+    private void printTotalNumberOfTasks() {
+        int numTasks = tasks.size();
+        if (numTasks < 2) {
```
Is there a better way to structure these to extract common code and reduce duplicates?

Zhang-Yunjie commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,64 @@
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
Would it be better to change "isExit" to "hasExit" to make it more natural?

Zhang-Yunjie commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+     *
+     * @return true or false
+     */
+    public abstract boolean isExit();
```
May I ask why this must be a method instead of a boolean attribute?

Zhang-Yunjie commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,38 @@
+     * @param storage storage storage that handles data storage
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
Would it be better to name it more specifically, e.g. "listOutTasks" instead of "execute" ?

Zhang-Yunjie reviewed (commented):
LGTM, almost. I like how you structure your classes very neatly so that it is easy to understand as someone new to your codes :) Good job 👍

hanweic53 committed:
Improve GUI

hanweic53 committed:
Merge branch 'branch-Level-10' into master

hanweic53 committed:
Add tag for A-Gradle

To satisfy the iP Progress Dashboard.
Something went wrong with the previous A-Gradle tagging.

fall9x commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,73 @@
+
+        try {
+            Command c = Parser.parse(input);
+            return "Duke:\n" + c.execute(taskList, ui, storage);
```
Perhaps you might want to move the string output to the ui class as well

xuey0ng commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,50 @@
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
```
Perhaps you might want add a condition to check if the user wants to exit, before using Platform.exit() to terminate the javaFx window.

xuey0ng commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,70 @@
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws IOException, DukeException {
+        try {
+            switch (commandWord) {
+            case "todo":
+                return taskList.addTodo(taskName, storage);
+
+            case "deadline":
+                return taskList.addDeadline(taskName, storage);
+
+            case "event":
+                return taskList.addEvent(taskName, storage);
+
+            default:
+                return "";
```
Nice job on the switch indentation!

fall9x commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,39 @@
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+
+        return "Bye. Hope to see you again soon!";
```
Perhaps you might want to move this to the Ui?

xuey0ng reviewed (commented):
I like the way you structure your classes! . Overall LGTM:)

hanweic53 committed:
Improve GUI and repackage classes

hanweic53 committed:
Add A-Assertions: Use Assertions

Wrote assumptions that should hold at various points in the code.

hanweic53 committed:
Add A-CodeQuality: Improve Code Quality

Examine code and refactor to improve code quality.

hanweic53 committed:
Merge pull request #1 from hanweic53/branch-A-Assertions

Add A-Assertions: Use Assertions

hanweic53 committed:
Merge branch 'master' into branch-A-CodeQuality

hanweic53 committed:
Merge pull request #2 from hanweic53/branch-A-CodeQuality

Add A-CodeQuality

hanweic53 committed:
Add A-CI: Set up CI

Use Continuous Integration.

hanweic53 committed:
Refactor code

Futher improvements on code quality.

hanweic53 committed:
Add C-Tagging

Allows tagging of tasks.

hanweic53 committed:
Merge branch 'branch-C-Tagging' into master

HynRidge reviewed (commented):
Hey Buddy! Your code looks clean and all things seem to be well abstracted.. Kudos

hanweic53 committed:
Improve Tagging functions

hanweic53 committed:
Add A-BetterGui: Better Gui and refactor code

hanweic53 committed:
Add A-UserGuide: User Guide

hanweic53 committed:
Set theme jekyll-theme-slate

hanweic53 committed:
Add TagList: Improve Tagging functionality

hanweic53 committed:
Set theme jekyll-theme-slate

hanweic53 committed:
Merge branch 'master' of https://github.com/hanweic53/ip into master

hanweic53 committed:
Set theme jekyll-theme-merlot

hanweic53 committed:
Set theme jekyll-theme-modernist

hanweic53 committed:
Edit README.md

hanweic53 committed:
Edit README.md

hanweic53 committed:
Edit Duke's welcome message

hanweic53 committed:
Include links in README

hanweic53 committed:
Update README.md
